### PR TITLE
debug: Improve namespace and region support

### PR DESCRIPTION
**Pull request by davemay99**
This PR provides additional insight into namespaces and regions known to the cluster.

* Add namespace/region filter to CLI output
* Add region filtering for servers
* Add namespaces.json and regions.json to cluster meta
* Add helper function to test for presence of string prefix in slice of strings
* Refactor tests to validate correct behavior

### CODE REVIEW

**Review by notnoop — COMMENTED** [review]
The usability improvement is pretty nice. I have few stylistic comments as well as a question about SliceStringContainsPrefix logic.

File: command/operator_debug.go [hunk 1]
```diff
@@ -1055,8 +1064,44 @@ func TarCZF(archive string, src, target string) error {
...
+		if helper.SliceStringContainsPrefix(prefixes, member.Name) {
+			membersFound = append(membersFound, member.Name)
+		}
+	}
+
+	return membersFound, nil
+}
+
+// stringToSlice splits CSV string into slice, trims whitespace, and prunes
+// empty values
```
[the remark this answers is not in the document]
Comment (davemay99) [reply command/operator_debug.go:1103]: Done

File: command/operator_debug_test.go [hunk 2]
```diff
@@ -73,13 +73,15 @@ func TestDebug_NodeClass(t *testing.T) {
...
 
 	// Wait for client1 to connect
-	client1NodeID := client1.Agent.Client().NodeID()
-	testutil.WaitForClient(t, srv.Agent.RPC, client1NodeID)
-	t.Logf("[TEST] Client1 ready, id: %s", client1NodeID)
+	client1 := agent1.Agent.Client()
+	client1NodeID := client1.NodeID()
+	client1Region := client1.Region()
+	testutil.WaitForClient(t, srv.Agent.RPC, client1NodeID, client1Region)
+	t.Logf("[TEST] Client1 ready, id: %s, region: %s", client1NodeID, client1Region)
```
[the remark this answers is not in the document]
Comment (davemay99) [reply command/operator_debug_test.go:84]: Good call - done!

File: command/operator_debug_test.go [hunk 3]
```diff
@@ -210,6 +218,88 @@ func TestDebug_ClientToServer(t *testing.T) {
 	runTestCases(t, cases)
 }
 
+func TestDebug_ClientToServer_Region(t *testing.T) {
+	agentConfFunc := func(c *agent.Config) {
+		c.Region = "testregion"
+	}
+
+	// Start test server and API client
+	srv, _, url := testServer(t, false, agentConfFunc)
```
[the remark this answers is not in the document]
Comment (davemay99) [reply command/operator_debug_test.go:227]: Makes sense -- fixed!

File: command/operator_debug_test.go [hunk 4]
```diff
@@ -432,15 +522,26 @@ func TestDebug_Fail_Pprof(t *testing.T) {
...
+		expected []string
+	}{
+		{input: ",,", expected: []string(nil)},
+		{input: "", expected: []string(nil)},
+		{input: "foo, bar", expected: []string{"foo", "bar"}},
+	}
+	for _, tc := range cases {
+		out := stringToSlice(tc.input)
+		require.Equal(t, tc.expected, out)
+		require.Equal(t, true, helper.CompareSliceSetString(tc.expected, out))
```
[the remark this answers is not in the document]
Comment (davemay99) [reply command/operator_debug_test.go:539]: Removed the second require, and added a couple more cases to verify trim/prune work as expected.

File: command/operator_debug.go [hunk 5]
```diff
@@ -1055,8 +1064,44 @@ func TarCZF(archive string, src, target string) error {
...
 
-// argNodes splits node ids from the command line by ","
-func argNodes(input string) []string {
+// parseMembers returns a slice of server member names matching the search criteria
+func parseMembers(serverMembers *api.ServerMembers, serverIDs string, region string) (membersFound []string, err error) {
+	prefixes := stringToSlice(serverIDs)
+
+	if serverMembers.Members == nil {
+		return nil, fmt.Errorf("Failed to parse server members, members==nil")
+	}
```
[the remark this answers is not in the document]
Comment (davemay99) [reply command/operator_debug.go:1073]: Done!

File: command/operator_debug.go [hunk 6]
```diff
@@ -1055,8 +1064,44 @@ func TarCZF(archive string, src, target string) error {
...
+		if serverIDs == "all" {
+			membersFound = append(membersFound, member.Name)
+			continue
+		}
+
+		// Special case passthrough as literally "leader"
+		if serverIDs == "leader" {
+			membersFound = append(membersFound, "leader")
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (davemay99) [reply command/operator_debug.go:1091]: Good catch!  I moved it above the loop and added a comment to describe the special case it covers.

File: helper/funcs.go [hunk 7]
```diff
@@ -197,6 +197,16 @@ func SliceStringContains(list []string, item string) bool {
 	return false
 }
 
+// SliceStringContainsPrefix returns true if any string in list matches prefix
+func SliceStringContainsPrefix(list []string, prefix string) bool {
```
[the remark this answers is not in the document]
Comment (davemay99) [reply helper/funcs.go:201]: The logic was indeed reversed.  I left this function in place and added StringHasPrefixInSlice to cover the expected behavior, and wrote tests for both.

File: helper/funcs_test.go [hunk 8]
```diff
@@ -34,6 +34,27 @@ func TestSliceStringContains(t *testing.T) {
...
+
+func TestStringHasPrefixInSlice(t *testing.T) {
+	prefixes := []string{"a", "b", "c", "definitely", "most definitely"}
+	require.True(t, StringHasPrefixInSlice("alpha", prefixes))
+	require.True(t, StringHasPrefixInSlice("bravo", prefixes))
+	require.True(t, StringHasPrefixInSlice("charlie", prefixes))
+	require.False(t, StringHasPrefixInSlice("delta", prefixes))
+	require.True(t, StringHasPrefixInSlice("definitely", prefixes))
+	require.False(t, StringHasPrefixInSlice("most", prefixes))
+	require.True(t, StringHasPrefixInSlice("most definitely", prefixes))
```
Comment (notnoop) [inline helper/funcs_test.go:55]: I'm finding it hard to digest the tests and expected results. Separating the True from false cases would help.

Comment (davemay99) [reply helper/funcs_test.go:55]: I separated the true/false cases, used common elements between the two tests, and added comments to help improve readability.

**Pull request merged**